src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http3.cs

```diff
@@ -1626,6 +1626,7 @@ public async Task DuplexStreaming_AbortByServer_StreamingCancelled(bool graceful
         public async Task ServerSendsTrailingHeaders_Success()
         {
             using Http3LoopbackServer server = CreateHttp3LoopbackServer();
+            SemaphoreSlim clientFinishedSemaphore = new SemaphoreSlim(0);
```

Comment:
This is ok, but for future reference a semaphore here is a bit overkill.  It could just be a TaskCompletionSource, for example.